### PR TITLE
feat(Library of Babel): add presence

### PR DESCRIPTION
**Pull request by ION606**
## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

image

image

</details>

### CODE REVIEW

File: websites/L/Library Of Babel/metadata.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+	"author": {
+		"id": "358402930191106049",
+		"name": "ion606"
+	},
+	"service": "Library Of Babel",
+	"description": {
+		"en": "A rich presence for libraryofbabel.info"
+	},
+	"url": "libraryofbabel.info",
+	"version": "1.0.0",
```
Comment (theusaf) [inline websites/L/Library Of Babel/metadata.json:12]:
```suggestion
	"url": [
		"libraryofbabel.info",
		"www.libraryofbabel.info"
	],
	"matches": [
		"*://libraryofbabel.info/*",
		"*://www.libraryofbabel.info/*"
	],
	"version": "1.0.0",
```

File: websites/L/Library Of Babel/metadata.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+	"logo": "https://libraryofbabel.info/img/hex802.png",
```
Comment (theusaf) [inline websites/L/Library Of Babel/metadata.json:13]: Logos should be 512x512px

File: websites/L/Library Of Babel/metadata.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+	"thumbnail": "https://libraryofbabel.info/img/hex802.png",
```
Comment (theusaf) [inline websites/L/Library Of Babel/metadata.json:14]: > The thumbnail should preferably be a [wide promotional card](https://i.imgur.com/3QfIc5v.jpg) or a [screenshot](https://i.imgur.com/OAcBmwW.png) if the first is not available.

File: websites/L/Library Of Babel/presence.ts [hunk 4]
```diff
@@ -0,0 +1,120 @@
+const presence = new Presence({
+	clientId: "1212508565802192936",
+});
+
+// Other default assets can be found at index.d.ts
+const enum Assets {
+	Logo = "https://libraryofbabel.info/img/hex802.png",
+	smallImageKey = "",
```
Comment (theusaf) [inline websites/L/Library Of Babel/presence.ts:8]: This asset doesn't appear to be used
```suggestion
```

File: websites/L/Library Of Babel/presence.ts [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+		case Pages.images:
+			presenceData.details = "viewing images";
+			break;
+
+		case Pages.coloroscopy:
+			presenceData.details = "undergoing coloroscopy";
+			break;
+
+		default:
+			// console.log(base);
```
Comment (theusaf) [inline websites/L/Library Of Babel/presence.ts:114]: Remove unused commented code

File: websites/L/Library Of Babel/presence.ts [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+		},
+		p = (document.querySelector("#page") as HTMLInputElement)?.value;
+
+	switch (base) {
+		case Pages.homepage:
+			presenceData.details = "in the lobby (home)";
+			break;
+
+		case Pages.about:
+			presenceData.details = "learning about the archives";
```
Comment (theusaf) [inline websites/L/Library Of Babel/presence.ts:52]: Typically, the strings in `details` are capitalized.
```ts
"Learning about the archives"
```

File: websites/L/Library Of Babel/presence.ts [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+			.replace("Anglishized", "")
+			.replace("Theory - ", "")
+			.replace("Image Search: ", ""),
+		presenceData: PresenceData = {
+			largeImageKey: Assets.Logo,
+			largeImageText: "hex802",
+			name: "Library of Babel",
+			type: ActivityType.Watching,
+		},
+		p = (document.querySelector("#page") as HTMLInputElement)?.value;
```
Comment (theusaf) [inline websites/L/Library Of Babel/presence.ts:44]: Use generics rather than `as Type`

File: websites/L/Library Of Babel/presence.ts [hunk 8]
```diff
@@ -0,0 +1,120 @@
...
+	imgsearch = "/imagesearch.html",
+	imgSRes = "/imagesearch.cgi",
+	coloroscopy = "/coloroscopy.html",
+	refhex = "/referencehex.html",
+}
+
+presence.on("UpdateData", async () => {
+	if (!document.location.toString()) return;
+
+	let base = window.location.pathname;
```
Comment (theusaf) [inline websites/L/Library Of Babel/presence.ts:31]:
```suggestion
	let base = document.location.pathname;
```

File: websites/L/Library Of Babel/presence.ts [hunk 9]
```diff
@@ -0,0 +1,120 @@
...
+	theory = "/theory.html",
+	images = "/slideshow.html",
+	imgsearch = "/imagesearch.html",
+	imgSRes = "/imagesearch.cgi",
+	coloroscopy = "/coloroscopy.html",
+	refhex = "/referencehex.html",
+}
+
+presence.on("UpdateData", async () => {
+	if (!document.location.toString()) return;
```
Comment (theusaf) [inline websites/L/Library Of Babel/presence.ts:29]: This condition is likely always false?

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/L/Library Of Babel/metadata.json (hunk 3 above)
Comment (ION606) [reply websites/L/Library Of Babel/metadata.json:21]: ohhhh I see, would something like this be better? 
image

File: websites/L/Library Of Babel/presence.ts (hunk 9 above)
Comment (ION606) [reply websites/L/Library Of Babel/presence.ts:28]: I added this to get past a weird this firefox does when it redirects to `about:blank`

**Comment by ION606** [comment]
should I....resolve the issues (I changed a bunch of them in the latest merge) @theusaf

File: websites/L/Library Of Babel/presence.ts [hunk 10]
```diff
@@ -0,0 +1,119 @@
+const presence = new Presence({
+	clientId: "1212508565802192936",
+});
+
+// Other default assets can be found at index.d.ts
+const enum Assets {
+	Logo = "https://libraryofbabel.info/img/hex802.png",
```
Comment (theusaf) [inline websites/L/Library Of Babel/presence.ts:7]:
```suggestion
const enum Assets {
	Logo = "https://i.imgur.com/e9ewOCy.png",
```

File: websites/L/Library Of Babel/presence.ts (hunk 8 above)
Comment (theusaf) [reply websites/L/Library Of Babel/presence.ts:31]: This repository prefers `document.location` for consistency

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/L/Library Of Babel/presence.ts [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+			presenceData.details = `Reading ${title.replace(
+				` ${p}`,
+				""
+			)} (page ${p})`;
+			break;
+
+		case Pages.browse:
+			{
+				// find the shelf and page (not added for now, perhaps implement later)
+				presenceData.details = "Browsing the archives";
```
Comment (DarkVillager) [inline websites/L/Library Of Babel/presence.ts:95]: Example how to implement this, just make those variables or sum
```suggestion
				presenceData.details = "Browsing the archives";
				presenceData.state = `Hex ${
	document.querySelector<HTMLTextAreaElement>("#hexer")?.value
} - Wall ${document
	.querySelector<HTMLSelectElement>("#inside1")
	?.value?.slice(1)} - Shelf ${document
	.querySelector<HTMLSelectElement>("#inside2")
	?.value?.slice(1)} - Volume ${document
	.querySelector<HTMLSelectElement>("#inside3")
	?.value?.slice(1)}`;
```
image

**Pull request closed without merging**